### ISSUE TITLE
Positive Learning Rate in Optimizer

### ISSUE THREAD

**Issue by akshay-raj-dhamija** [facebookarchive/caffe2#769]
Hi,
I am using the following code for an sgd solver
```
optimizer.build_sgd(model, base_learning_rate=0.1, policy="step", stepsize=1, gamma=0.999)
```
I use `workspace.FetchBlob('conv1_b_lr')` to fetch the learning rate for every iteration and that results in the plot below. But, its starting point is -0.1 rather than 0.1 hence giving an illusion of rising learning rate.
![image](https://user-images.githubusercontent.com/13166397/26999492-070a4584-4d58-11e7-9693-23a7dd22440d.png)
I believe, the learning rate is being returned with an opposite sign.

**Comment by mfawzymkh** [comment]
@akshay-raj-dhamija do you still have this problem?  The - sign happens if you use WeightedSum (momentum) with SGD.

**Comment by akshay-raj-dhamija** [comment]
Thanks @mfawzymkh for your response.
I checked with the latest commit and the issue still exists. I am quite sure I am only using SGD.
The entire code is at http://vast.uccs.edu/~adhamija/blog/MNIST_singleGPU.html. Here, I multiply the learning rate by -1 to obtain the correct graph in cell 24.
Please do let me know in case something seems incorrect.

**Comment by akshay-raj-dhamija** [comment]
Hi @mfawzymkh, were you able to narrow down the issue. Is it my usage?
